ladybug_geometry/geometry3d/cone.py

```diff
@@ -0,0 +1,191 @@
+
+    @property
+    def area(self):
+        """Surface of a cone"""
+        return math.pi * self.radius ** 2 + math.pi * self.radius * self.slant_height
+
+    @property
+    def volume(self):
+        """Volume of a cone"""
+        return 1 / 3 * math.pi * self.radius ** 2 * self.height
+
+    def move(self, moving_vec):
+        """Get a cone that has been moved along a vector.
+
+        Args:
+            moving_vec: A Vector3D with the direction and distance to move the cone.
+        """
+        return Cone3D(self.vertex.move(moving_vec), self.axis, self.angle, self.height)
+
+    def rotate(self, axis, angle, origin):
+        """Rotate this cone by a certain angle around an axis and origin.
+
+        Right hand rule applies:
+        If axis has a positive orientation, rotation will be clockwise.
+        If axis has a negative orientation, rotation will be counterclockwise.
+
+        Args:
+            axis: A Vector3D axis representing the axis of rotation.
+            angle: An angle for rotation in radians.
+            origin: A Point3D for the origin around which the sphere will be rotated.
+        """
+        return Cone3D(self.vertex.rotate(axis, angle, origin),
+                      self.axis.rotate(axis, angle, origin),
+                      self.angle,
+                      self.height)
+
+    def rotate_xy(self, angle, origin):
+        """Get a cone that is rotated counterclockwise in the world XY plane by an angle.
+
+        Args:
+            angle: An angle for rotation in radians.
+            origin: A Point3D for the origin around which the sphere will be rotated.
+        """
+        return Cone3D(self.vertex.rotate_xy(angle, origin),
+                      self.axis.rotate_xy(angle, origin),
+                      self.angle,
+                      self.height)
+
+    def reflect(self, normal, origin):
+        """Get a cone reflected across a plane with the input normal vector and origin.
+
+        Args:
+            normal: A Vector3D representing the normal vector for the plane across
+                which the arc will be reflected. THIS VECTOR MUST BE NORMALIZED.
+            origin: A Point3D representing the origin from which to reflect.
+        """
+        return Cone3D(self.vertex.reflect(normal, origin),
+                      self.axis.reflect(normal, origin),
+                      self.angle,
+                      self.height)
+
+    def scale(self, factor, origin=None):
+        """Scale a cone by a factor from an origin point.
+
+        Args:
+            factor: A number representing how much the cone should be scaled.
+            origin: A Point3D representing the origin from which to scale.
+                If None, it will be scaled from the World origin (0, 0, 0).
+        """
+        return Cone3D(self.origin.scale(factor, origin),
+                      self.axis,
+                      self.angle,
+                      self.height)
```

Comment:
You will have to scale the height of the cone in addition to moving the vertex. Or rather, you'll have to scale the axis vector (after you make the change noted above).

Reply:
Done!